modules/MapTile.js

```diff
@@ -0,0 +1,122 @@
+
+//static bitflag defines for our directions
+const NORTH = 1;
+const SOUTH = 2;
+const EAST = 4;
+const WEST = 8;
+
+const FLOOR_A = 1;
+const FLOOR_B = 2;
+const FLOOR_C = 3;
+const FLOOR_D = 4;
+
+const FLOOR_RENDER_DEPTH = -0.000001;
+const WALL_RENDER_DEPTH = 0;
+
+class MapTile
+{
+	static scene;
+	static wallSpriteSource;
+	static wallSpriteDims;
+	static wallTileDims;
+	//static wallTileDimsRelative;
+	static dirLookupTileIndex;
+	static dirLookupTileDims;
+	static dirLookupFloorIndex;
+	static worldTileDefaults = new THREE.Vector2(32,32);
+	constructor(wallDir, tileCoords, floorID)
+	{
+		this.myDir = wallDir;
+		this.myCoords = tileCoords;
+		//look up the tile index from the chosen wall direction... it is stored as a string
+		//sprite sheet tile index starts from 0 in the bottom left corner
+		//this.myTileIndex = new THREE.Vector2(1, 21);	
+		let tileIndex;
+		if(MapTile.dirLookupTileIndex.has("" + wallDir))
+		{
+			tileIndex = MapTile.dirLookupTileIndex.get("" + wallDir);
+		}
+		else
+		{
+			//todo: draw a big red X or something to say broken sprite
+			tileIndex = new THREE.Vector2(36,21);
+		}
+		
+		//does this tile index have special dimensions handling? for any NORTH flag walls
+		let customWallTileDims = MapTile.wallTileDims
+		if(MapTile.dirLookupTileDims.has("" + wallDir))
+		{
+			 customWallTileDims = MapTile.dirLookupTileDims.get("" + wallDir);
+		}
+		const wallTileDimsRelative = new THREE.Vector2(
+			customWallTileDims.x/MapTile.wallSpriteDims.x,
+			customWallTileDims.y/MapTile.wallSpriteDims.y);
+		
+		//setup the render texture from the sprite sheet
+		this.myTexture = new THREE.Texture();
+		this.myTexture.source = MapTile.wallSpriteSource;
+		this.myTexture.repeat.set(wallTileDimsRelative.x, wallTileDimsRelative.y);		
+		this.myTexture.offset.set(tileIndex.x * wallTileDimsRelative.x, tileIndex.y * wallTileDimsRelative.y);
+		this.myTexture.needsUpdate = true;
```

Comment:
This part is also called more than once.

Reply:
This and the other bit was the 4am hack to get it working out of frustration

Reply:
This bit also has the functionality done twice but splitting it off into a function doesn't do much as it wouldnt be more efficient or save much line space